Frontpanel over ip milestones

hydrozoan opened the pull request:
Update to add the latest supported FrontPanel over IP methods and protocol changes.

The following items should be tested to verify they function as expected for connected devices that support the features and those that do not support the features.

- [ ] Flash Read/Write/Erase Sector
- [ ] Device Settings Set/Get, Delete, Save, List
- [ ] Device Sensors Get
- [ ] Configure FPGA with Reset Profile
- [ ] Configure FPGA from Flash
- [ ] ClearFPGAConfiguration
- [ ] Set/Get Reset Profile
- [ ] Set/Get PLL22150 Configuration
- [ ] LoadDefaultPLLConfiguration
- [ ] Set/Get Eeprom PLL22150 Configuration
- [ ] Read/Write Registers
- [ ] GetDeviceInfo provides HasDeviceSettingsSupport and HasDeviceSensorsSupport

hydrozoan committed:
Add Flash Read/Write/Erase methods.

hydrozoan committed:
Add FrontPanel Client class to maintain connection to a FrontPanel over IP Server.
This allows the possibility of passing the instance to other classes that need to communicate with the Server.

hydrozoan committed:
Add DeviceSettings class to provide access to Settings on a Device.

hydrozoan committed:
Add ConfigureFPGAWithReset and ConfigureFPGAFromFlash
Add ClearFPGAConfiguration
Add Get/Set ResetProfile

hydrozoan committed:
Add Get/Set device PLL22150 configuration.

hydrozoan committed:
Add regression tests for Device Flash.
Add regression tests for Device Sensors.
Add regression tests for Device Settings.
Add regression tests for PLL22150 Configuration.
Add regression tests for Reset Profiles.

hydrozoan committed:
Add Read/Write Registers.

hydrozoan committed:
Add regression tests for Read/Write Registers.

hydrozoan committed:
Update to use the latest version of the DeviceInfo structure that added isDeviceSettingsSupported and isDeviceSensorsSupported

hayhut commented:
@hydrozoan Could you make a checklist in the PR description with all the things you want tested? 
- [ ] foo

hydrozoan commented:
> @hydrozoan Could you make a checklist in the PR description with all the things you want tested?
> 
> * [ ]  foo

I updated the description with a list of items that should guide the testing. I tested much of this with the regression tests but it would be nice to also test it by some other means. There is some code in the repository for an application that runs in the browser but it would need to be extended to use the new functions. What are your thoughts on how you plan to test this?

hydrozoan committed:
Update to correct and improve comments for API functions.

hydrozoan committed:
Update dependencies using npm audit.

hydrozoan committed:
Fix errors resulting from updating dependencies.

hydrozoan committed:
Update with changes to the FrontPanel over IP protocol for opening a device and retrieving the private DeviceInfo.
Add functions to determine if DeviceSettings and or DeviceSensors are supported.

hydrozoan committed:
Update the README to show the correct command to install the npm package for the FrontPanel WebAPI.

hydrozoan committed:
Update the ProductID enumeration to match that of the FrontPanelAPI.

hydrozoan committed:
Update the frontpanel-ws.js file in the sample webapp.

hydrozoan committed:
Increment version number to 0.2.4

hydrozoan committed:
Increment version to 0.2.5 for republish.

hydrozoan committed:
Add Get DeviceSensors to retrieve sensor data from device.

hydrozoan committed:
Update dependencies using npm update.

hydrozoan committed:
Increment WebAPI version to 0.2.6

hydrozoan committed:
Update the FrontPanel WebAPI to the latest 0.2.6 version.

hydrozoan committed:
Update Browser Sample JQuery.js version.

hydrozoan committed:
Update Bootstrap dependencies for Browser Sample.
